actors/builtin/miner/miner_state.go

```diff
@@ -761,11 +763,31 @@ func (st *State) PenalizeFundsInPriorityOrder(store adt.Store, currEpoch abi.Cha
 
 	// unlocked funds were just deducted from available, so track that
 	remaining := big.Sub(target, fromVesting)
-
 	fromBalance = big.Min(unlockedBalance, remaining)
+
+	// track unpaid fee in fee debt field
+	remaining = big.Sub(remaining, fromBalance)
+	if remaining.GreaterThan(big.Zero()) {
+		st.FeeDebt = big.Add(st.FeeDebt, remaining)
+	}
+
 	return fromVesting, fromBalance, nil
 }
 
+// Repays the full miner actor fee debt.  Returns the amount that must be
+// repaid and an error if there are not sufficient funds to cover repayment.
+// Miner state repays from unlocked funds, potentially violating IP requirements
+// and bringing actor into IP debt.  FeeDebt should be  zero after calling.
+func (st *State) RepayDebt(currBalance abi.TokenAmount) (abi.TokenAmount, error) {
+	unlockedBalance := st.GetUnlockedBalance(currBalance)
+	if !st.CanRepayFeeDebt(unlockedBalance) {
+		return big.Zero(), xerrors.Errorf("unlocked balance can not repay fee debt (%v < %v)", unlockedBalance, st.FeeDebt)
```

Comment:
I think this is an appropriate place for `exitcode.ErrInsufficientFunds.Wrapf("unlock....")`